zilliqa/src/cfg.rs

```diff
@@ -659,20 +684,70 @@ impl ContractUpgradesBlockHeights {
                     if v.is_null() {
                         None // Skip null values
                     } else {
-                        Some((k, toml::Value::Integer(v.as_u64().unwrap() as i64)))
+                        Some((
+                            k,
+                            toml::Value::Table(
+                                // ContractUpgradeConfig
+                                json!(v)
+                                    .as_object()
+                                    .unwrap()
+                                    .clone()
+                                    .into_iter()
+                                    .filter_map(|(k2, v2)| {
+                                        if v2.is_u64() {
```

Comment:
This feels extremely confusing and error-prone. What exactly are we trying to achieve with this function? Surely there's a saner way?

Reply:
Wow that is bad. Pushed improvement